code/modules/mob/mob_grab.dm

```diff
@@ -62,6 +64,55 @@
 	clean_grabbed_by(assailant, affecting)
 	adjust_position()
 
+/obj/item/grab/Destroy()
+	if(affecting)
+		UnregisterSignal(affecting, COMSIG_MOVABLE_MOVED)
+		if(!affecting.buckled)
+			affecting.pixel_x = 0
+			affecting.pixel_y = 0 //used to be an animate, not quick enough for qdel'ing
+			affecting.layer = initial(affecting.layer)
+		affecting.grabbed_by -= src
+		affecting = null
+	if(assailant)
+		UnregisterSignal(assailant, COMSIG_MOVABLE_MOVED)
+		if(assailant.client)
+			assailant.client.screen -= hud
+		assailant = null
+	QDEL_NULL(hud)
+	return ..()
+
+/obj/item/grab/proc/pull_grabbed(mob/user, turf/old_turf, direct, forced)
+	if(assailant.moving_diagonally == FIRST_DIAG_STEP) //we dont want to do anything in the middle of diagonal step
+		return
+	if(!assailant.Adjacent(old_turf))
+		qdel(src)
+		return
+	if(assailant.Adjacent(affecting))
+		adjust_position()
+		return
+	var/possible_dest = list(old_turf)
+	for(var/turf/dest in orange(assailant, 1))
+		if(dest.Adjacent(affecting))
+			possible_dest |= dest
+	if(istype(assailant.l_hand, /obj/item/grab))
+		var/obj/item/grab/grab = assailant.l_hand
+		possible_dest -= get_turf(grab.affecting)
+	if(istype(assailant.r_hand, /obj/item/grab))
+		var/obj/item/grab/grab = assailant.r_hand
+		possible_dest -= get_turf(grab.affecting)
+	for(var/turf/dest in possible_dest)
```

Comment:
```suggestion
	for(var/turf/dest as anything in possible_dest)
```
It only has turfs in the list. So `as anything` will make it so that Byond won't apply a filter on the turf type, slightly faster